### ISSUE TITLE
`no-return-await` does not work well for boolean expression

### ISSUE THREAD

**Issue by wanganjun** [palantir/tslint#3613]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
async function bool1(): Promise<boolean> {
    return true;
}

async function bool2(): Promise<boolean> {
    return false;
}

async function test(): Promise<boolean> {
    return (await bool1()) && (await bool2());
}
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "jsRules": {},
    "rules": {
        "no-return-await": true,
        "no-unsafe-finally": true,
        "promise-function-async": true,
        "quotemark": [true, "single", "avoid-escape", "avoid-template"],
        "semicolon": [true, "always"]
    },
    "rulesDirectory": []
}
```

#### Actual behavior
the second `await` in function `test` will be considered as unnecessary.

#### Expected behavior
there should be no `no-return-await` linting error for function `test`
